k9mail-library/src/main/java/com/fsck/k9/mail/store/imap/ImapFolder.java

```
@@ -1248,17 +1248,40 @@ public String getUidFromMessageId(Message message) throws MessagingException {
     }
 
     @Override
-    public void expunge() throws MessagingException {
+    public void expunge(List<Long> knownDeletedUids) throws MessagingException {
+        if (knownDeletedUids == null || knownDeletedUids.isEmpty()) {
```

Both cases are programming errors. So I'd rather throw an exception than returning silently.